onnxruntime/python/tools/quantization/quant_utils.py

```diff
@@ -527,7 +528,12 @@ def model_has_infer_metadata(model):
 
 
 def load_model_with_shape_infer(model_path: Path):
-    inferred_model_path = generate_identified_filename(model_path, "-inferred")
+    temp_name_id = 0
+    while True:
+        temp_name_id += 1
+        inferred_model_path = generate_identified_filename(model_path, f"-inferred{time.time()}.{temp_name_id}")
+        if not inferred_model_path.exists():
+            break
```

Comment:
```suggestion
    for i in itertools.count():
        inferred_model_path = generate_identified_filename(model_path, f"-inferred{time.time()}.{i}")
        if not inferred_model_path.exists():
            break
```
Avoid "temp" in variable names; avoid while True loops

Helpful to also explain in a comment why this name gen process is needed?

Reply:
changing it to tempfile